extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/discovery/SceneDiscoveryService.java

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.digitalstrom.internal.discovery;
+
+import static org.eclipse.smarthome.binding.digitalstrom.DigitalSTROMBindingConstants.*;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.smarthome.binding.digitalstrom.handler.BridgeHandler;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.scene.InternalScene;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.scene.constants.SceneEnum;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.structure.scene.constants.SceneTypes;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Sets;
+
+/**
+ * The {@link SceneDiscoveryService} discovers all digitalSTROM-scene of one scene-type. The scene-type have to be
+ * given to the {@link SceneDiscoveryService#SceneDiscoveryService(BridgeHandler, ThingTypeUID)} as
+ * {@link ThingTypeUID}, which has a scene-type as ThingType id. The scene types can be found in {@link SceneTypes}.
+ *
+ * @author Michael Ochel - Initial contribution
+ * @author Matthias Siegele - Initial contribution
+ */
+public class SceneDiscoveryService extends AbstractDiscoveryService {
+
+    private final static Logger logger = LoggerFactory.getLogger(SceneDiscoveryService.class);
+
+    private BridgeHandler digitalSTROMBridgeHandler;
+
+    private String sceneType;
+
+    /**
+     * Creates a new {@link SceneDiscoveryService} for the given supportedThingType, which has to be a SceneThing with
+     * one of the valid {@link SceneTypes}.
+     *
+     * @param digitalSTROMBridgeHandler - mandatory
+     * @param supportedThingType - mandatory
+     * @throws IllegalArgumentException
+     */
+    public SceneDiscoveryService(BridgeHandler digitalSTROMBridgeHandler, ThingTypeUID supportedThingType)
+            throws IllegalArgumentException {
+        super(Sets.newHashSet(supportedThingType), 60, false);
+        this.sceneType = supportedThingType.getId();
+        this.digitalSTROMBridgeHandler = digitalSTROMBridgeHandler;
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
```

Comment:
remove this method as it will never be called for a discovery service that does not support background discovery.